### ISSUE TITLE
Add ability to crossfade slides

### ISSUE THREAD

**Issue by stefanschleich** [fritzmg/contao-swiper#7]
Swiper has the ability to crossfade slides instead of simply fading them. This is useful if you fade any other elements than images, because with the simple fade elements behind the current slide are visible through the active slide.

The option can be set with the following parameter when initializing Swiper:

```javascript
fadeEffect: {
    crossFade: true
}
```

Source: https://idangero.us/swiper/api/#fade-effect

I'm not sure though if there should be a new effect (e.g. 'Crossfade') available in the dropdown of the element or a checkbox that becomes visible when the effect 'Fade' is chosen.

As a workaround I appended the parameter to the parameters-array in ce_swiperStop.html5:

```php
<?php
    $swiperParams = $this->parameters;
    $swiperParams['fadeEffect']['crossFade'] = 'true';
?>

<script>
    initSwiper({
        id: '<?= $this->sliderId ?>',
        wrapperClass: '<?= $this->wrapperClass ?>',
        params: <?= json_encode($swiperParams) ?>
    });
</script>
```

**Comment by fritzmg** [comment]
Do you have an example somewhere online, where one can see the difference with and without that option (when setting `effect` to `fade`)?

But yeah, in general Swiper has a lot more features, that cannot yet be enabled through the back end. The plan is to add more over time - whenever I or someone else needs them ;)

**Comment by fritzmg** [comment]
Btw. you could also do this:
```html
<script>
    var params = <?= json_encode($swiperParams) ?>;
    params.fadeEffect = {
        crossFade: true
    };
    initSwiper({
        id: '<?= $this->sliderId ?>',
        wrapperClass: '<?= $this->wrapperClass ?>',
        params: params
    });
</script>
```
Either way is fine.

**Comment by fritzmg** [comment]
I am still unable to see any difference in the fade effect. Do you have an example?

**Comment by stefanschleich** [comment]
Here are two examples:

Simple fade with text elements overlaying:
https://jsfiddle.net/5wtrve01/

Crossfade with text elements not overlaying:
https://jsfiddle.net/30vnhow4/

**Comment by fritzmg** [comment]
Ah, I see now. I always only tried with images - and then it does not make any difference at all.

But what is the use case for the first case? Do you ever want `fade` without `crossFade: true`?

**Comment by stefanschleich** [comment]
> Do you ever want fade without crossFade: true?

Yes, because when crossfading images the current one fades to transparent and only after that the new one fades in. So during the fade you see the background of the element the images are placed in. Hard to put into words, just try it with images and you'll see the difference.

**Comment by fritzmg** [comment]
> Yes, because when crossfading images the current one fades to transparent and only after that the new one fades in.

Hm, I cannot confirm that on my system. Can you show me an example again? ;)

**Comment by stefanschleich** [comment]
https://jsfiddle.net/sv0r46tp/

**Comment by stefanschleich** [comment]
Oh and to make it a bit more confusing: I had the problem too using transparent images (SVG in my case), so you can't say one is right for text and one for images. It depends on the content.

**Comment by fritzmg** [comment]
Hm, it does not do exactly as you described it, but I see what you mean now. Without crossfade, the slide behind the current slide stays fully opaque, while only the front slide is fading away. With crossfade, the active slide is fading away, while at the same time the next slide is fading in.

This means that during the transition, neither slides will be fully opaque and thus the background will be blended in as well.

I'd suggest to make it just another option, not a checkbox. So you have `Fade` & `Crossfade`. Should be more user friendly.

**Comment by stefanschleich** [comment]
Sounds good to me 👍

**Comment by fritzmg** [comment]
@stefanschleich could you please try `dev-master`?

**Comment by stefanschleich** [comment]
Works like a charm.

**Comment by fritzmg** [comment]
Released as `3.1.0` :)
